crates/oxc_linter/src/rules/vitest/prefer_spy_on.rs

```diff
@@ -0,0 +1,324 @@
+        .with_label(span)
+}
+
+#[derive(Debug, Default, Clone)]
+pub struct PreferSpyOn;
+
+declare_oxc_lint!(
+    /// ### What it does
+    ///
+    /// Enforce that all mocking is done using `vi.spyOn`.
+    ///
+    /// ### Why is this bad?
+    ///
+    /// Directly reassigning properties (e.g., obj.fn = vi.fn()) can break getters/setters,
+    /// lose original property descriptors, and fail with ESM read-only bindings.
+    /// Using `vi.spyOn()` makes the mocked property explicit, preserves the original descriptor,
+    /// and provides a built-in API to restore the original method without manually backing it up.
+    ///
+    /// ### Examples
+    ///
+    /// Examples of **incorrect** code for this rule:
+    /// ```js
+    /// Date.now = vi.fn()
+    /// Date.now = vi.fn(() => 10)
+    /// ```
+    ///
+    /// Examples of **correct** code for this rule:
+    /// ```js
+    /// vi.spyOn(Date, 'now')
+    /// vi.spyOn(Date, 'now').mockImplementation(() => 10)
+    /// ```
+    PreferSpyOn,
+    vitest,
+    style,
+    suggestion,
+);
+
+fn is_vi_fn_call<'a>(node: &Expression<'a>) -> bool {
+    match node {
+        Expression::StaticMemberExpression(static_expression) => {
+            is_vi_fn_call(&static_expression.object)
+        }
+        Expression::CallExpression(call) => call.callee.is_specific_member_access("vi", "fn"),
+        _ => false,
+    }
+}
+
+fn get_callee<'a>(node: &'a Expression<'a>) -> Option<&'a Expression<'a>> {
+    match node {
+        Expression::CallExpression(call) => Some(&call.callee),
+        _ => None,
+    }
+}
+
+fn find_node_object<'a>(node: &'a Expression<'a>) -> Option<&'a Expression<'a>> {
+    if let Some(callee) = get_callee(node) {
+        return find_node_object(&callee);
+    }
+
+    if let Some(member) = node.as_member_expression() {
+        return Some(member.object());
+    }
+
+    return None;
+}
+
+fn get_vitest_fn_call<'a>(node: &'a Expression<'a>) -> Option<&'a Expression<'a>> {
+    let call = node.is_call_expression() || node.is_member_expression();
+
+    if !call {
+        return None;
+    }
+
+    let Some(object) = find_node_object(node) else {
+        return None;
+    };
+
+    match object {
+        Expression::Identifier(_) => {
+            let value = {
+                let is_call_expression = node.is_call_expression() && is_vi_fn_call(node);
+                if is_call_expression { Some(node) } else { None }
+            };
+            return value;
+        }
+        _ => get_vitest_fn_call(object),
+    }
+}
+
+fn parent_has_mock_implementation_call<'a>(assignment: &'a Expression<'a>) -> bool {
+    let Some(outer_callee) = get_callee(assignment) else {
+        return false;
+    };
+
+    let Some(static_member) = outer_callee.as_member_expression() else {
+        return false;
+    };
+
+    let Some(property_name) = static_member.static_property_name() else {
+        return false;
+    };
+
+    property_name == "mockImplementation"
+}
```

Comment:
This method try to replicate this exact condition https://github.com/vitest-dev/eslint-plugin-vitest/blob/main/src/rules/prefer-spy-on.ts#L46.

As I didn't find any method to go up, I decide to go from up to down. After debugging the AST structure I found the following structure:
```
   CallExpression(
      Callee: StaticMemberExpresion(
                  object: CallExpression ( ---> vi.fn())
                  property: Identifier ( name: `mockImplementation`)
```